draft-ietf-masque-connect-udp-listen.md

```diff
@@ -164,6 +166,36 @@ and therefore is to be ignored). This document does not define any parameters
 for the Connect-UDP-Bind header field value, but future documents might define
 parameters. Receivers MUST ignore unknown parameters.
 
+
+# The Proxy-Public-Address Response Header Field {#response}
+
+Upon accepting the request, the proxy MUST select at least one public IP
+address to bind. The proxy MAY also select a second address from a different address
+family. For each selected address, it MUST select an open port to
+bind to this request. From then and until the tunnel is closed, the proxy
+SHALL send packets received on these IP-port tuples to the client. The proxy
+then MUST communicate the selected addresses and ports to the client using
+the "Proxy-Public-Address" header. The format of that header is defined below
```

Comment:
Rather than using a new header, can we add a parameter to the existing proxy-status header?

Reply:
Do we have a precedence of a Proxy-Status parameter being a MUST send? I could be incorrect, but I had the impression that Proxy-Status provided visibility rather than required behavior.

Not that I have a strong opinion, but to me use of a different field sounds more natural considering that the address MUST be included in the response.

Reply:
I'd be curious to hear what the benefit would be? If I understood your comment in the session correctly it sounds like you were suggesting it would be less confusing? But I am having a hard time differentiating it from "same outcome, different spelling"?

Reply:
At the session, there was confusion about this new proposed header being about "here are all of the proxy's public addresses", which it is not (it is "here are the addresses I'm using for this one request").

Proxy-Status is an established header that already tells the client about what the proxy did related to this one request, and already gives you remote IP addresses and other information. So adding this to there makes sense.

Regarding if it needs to be sent... I think it's OK for this document to say "if udp-bind clients need to get the local address, then this header needs to be sent"

Reply:
How do you determine if the bind clients need the address? That would have to be a flag in the initial client request

Reply:
No, I mean that the spec just says "clients need this, so you have to send it to them"